### DISCUSSION TITLE
How to stop at end of queue?

### DISCUSSION THREAD

**Discussion by wkuhns** [MusicPlayerDaemon/MPD#1916] [Q&A]
I'm a developer, but new to MPD. I've successfully integrated it with an existing database / music application, and it works perfectly except for one annoying behavior: at the end of the queue, MPD keeps playing songs. Is there an option to get it to simply stop when the queue is exhausted? I've searched through all the documentation I can find with no success.

**Comment by naglis** [comment]
Try [disabling `repeat`](https://mpd.readthedocs.io/en/latest/protocol.html#command-repeat) mode.

**Reply by wkuhns** [reply]
No joy. According to the docs, that option determines whether MPD will repeat a playlist (queue?) or a single song. In this case, it plays and consumes the loaded queue, then starts playing songs at random. Here's a screen dump of relevant MPC commands:

root@medialounge:/var/lib/mpd/playlists# mpc playlist
Loudon Wainwright III - Red Guitar
root@medialounge:/var/lib/mpd/playlists# mpc repeat 0
Loudon Wainwright III - Red Guitar
[playing] #1/1   0:21/1:51 (18%)
Updating DB (#1) ...
volume: n/a   repeat: off   random: off   single: off   consume: on 
root@medialounge:/var/lib/mpd/playlists# mpc seek 1:45
Loudon Wainwright III - Red Guitar
[playing] #1/1   1:45/1:51 (94%)
Updating DB (#1) ...
volume: n/a   repeat: off   random: off   single: off   consume: on 
root@medialounge:/var/lib/mpd/playlists# mpc playlist
Sean Paul - Like Glue

**Comment by MaxKellermann** [comment]
I don't get it. Stopping at the end of the queue is the default behavior (unless, as @naglis said, repeat mode is enabled, which is an option to explicitly change this very detail).
Your paste looks like some other MPD client keeps adding songs to your queue. Run MPD in verbose mode, and you'll see all commands issued by all clients, and then you'll see which one does this.

**Reply by wkuhns** [reply]
Thanks. Verbose mode didn't help identify the extra client, but I could see that there was one. It turns out that I also had mympd installed. Disabling mympd solved it.

**Reply by jcorporation** [reply]
The jukebox mode of myMPD adds random songs just before the last song of the queue ends, you can simply disable the jukebox mode from the myMPD ui.

**Reply by wkuhns** [reply]
Thanks again. I had the cli version of mympd, and I don't need any of its functionality in any event. Works perfectly now. I can fire it up again if I decide that it does something that I need.

**Reply by jcorporation** [reply]
There is no cli version of mympd ;)
